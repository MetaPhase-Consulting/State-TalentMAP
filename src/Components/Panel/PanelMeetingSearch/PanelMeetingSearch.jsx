import PropTypes from 'prop-types';
import SelectForm from 'Components/SelectForm';
import { PANEL_MEETINGS_PAGE_SIZES, PANEL_MEETINGS_SORT } from 'Constants/Sort';
import ExportButton from 'Components/ExportButton';
import { useState } from 'react';
import { panelMeetingsExport } from 'actions/availableBidders'; // Replace with correct Action file later

const PanelMeetingSearch = ({ isCDO }) => {
  const text = isCDO ? 'yes CDO' : 'no AO';
  // const exportDisabled = true;
  const [exportIsLoading, setExportIsLoading] = useState(false);
  const [limit, setLimit] = useState(PANEL_MEETINGS_PAGE_SIZES.defaultSize);
  const [ordering, setOrdering] = useState(PANEL_MEETINGS_SORT.defaultSort);

  const pageSizes = PANEL_MEETINGS_PAGE_SIZES;
  const sorts = PANEL_MEETINGS_SORT;

  const getQuery = () => ({
    limit,
    ordering,
  });

  const exportPanelMeetings = () => {
    if (!exportIsLoading) {
      setExportIsLoading(true);
      panelMeetingsExport(getQuery())
        .then(() => {
          setExportIsLoading(false);
        })
        .catch(() => {
          setExportIsLoading(false);
        });
    }
  };

  return (
    <div>
      Panel Meeting Search Page
      <div>
        Headers/Filters TBD
        isCDO: {text}
      </div>
      <div className="panel-results-controls-right">
        <div className="panel-results-controls">
          <SelectForm
            className="panel-results-select"
            id="panel-search-results-sort"
            options={sorts.options}
            label="Sort by:"
            defaultSort={ordering}
            onSelectOption={value => setOrdering(value.target.value)}
            // disabled={isLoading}
          />
          <SelectForm
            className="panel-results-select"
            id="panel-search-num-results"
            options={pageSizes.options}
            label="Results:"
            defaultSort={limit}
            onSelectOption={value => setLimit(value.target.value)}
            // disabled={isLoading}
          />
          <div className="export-button-container">
            <ExportButton
              onClick={exportPanelMeetings}
              isLoading={exportIsLoading}
              // disabled={exportDisabled}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

PanelMeetingSearch.propTypes = {
  isCDO: PropTypes.bool,
};

PanelMeetingSearch.defaultProps = {
  isCDO: false,
};

export default PanelMeetingSearch;
