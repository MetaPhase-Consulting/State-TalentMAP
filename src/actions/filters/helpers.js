import { getPostName } from '../../utilities';
import { COMMON_PROPERTIES } from '../../Constants/EndpointParams';
import { checkFlag } from '../../flags';

const getUseAP = () => checkFlag('flags.available_positions');
// Attempt to map the non-numeric grade codes to a full description.
// If no match is found, return the unmodified code.
export function getCustomGradeDescription(gradeCode) {
  switch (gradeCode) {
    case 'CM':
      return 'CM Career Minister (FE-CM)';
    case 'MC':
      return 'MC Minister-Counselor (FE-MC)';
    case 'OC':
      return 'OC Counselor (FE-OC)';
    case 'OM':
      return 'Office Manager (OM)';
    default:
      return gradeCode;
  }
}

function getLanguageNameByIfNull(filterItemObject = {}) {
  return filterItemObject.code === COMMON_PROPERTIES.NULL_LANGUAGE ?
    filterItemObject.customDescription || filterItemObject.formal_description
    :
    `${filterItemObject.formal_description} (${filterItemObject.code})`;
}

function getFuncRegionCustomDescription(shortDescription, longDescription) {
  return getUseAP() ? `(${shortDescription}) ${longDescription}` : false;
}

function getRegionCustomDescription(shortDescription, longDescription) {
  return `(${shortDescription}) ${longDescription}`;
}

function getSkillCustomDescription(description, code) {
  return `${description} (${code})`;
}

// create a custom description based on the filter type
// eslint-disable-next-line complexity
export function getFilterCustomDescription(filterItem, filterItemObject) {
  const { item: { description: descriptionPrimary } } = filterItem;
  const { short_description: shortDescription, long_description: longDescription, description,
    code, name } = filterItemObject;
  switch (descriptionPrimary) {
    case 'region':
      return getRegionCustomDescription(shortDescription, longDescription);
    case 'functionalRegion':
      return getFuncRegionCustomDescription(shortDescription, longDescription);
    case 'skill':
      return getSkillCustomDescription(description, code);
    case 'post':
      return getPostName(filterItemObject);
    case 'bidCycle':
      return name;
    case 'language':
      // language code NONE gets displayed differently
      return getLanguageNameByIfNull(filterItemObject);
    case 'grade':
      return getCustomGradeDescription(code);
    case 'postDiff': case 'dangerPay': case 'bidSeason':
      return description;
    default:
      return false;
  }
}

// Our standard method for getting a pill description.
// Pass a customType string for special rendering.
export function getPillDescription(filterItemObject, customType) {
  switch (customType) {
    case 'dangerPay':
      return `Danger pay: ${filterItemObject.description}`;
    case 'postDiff':
      return `Post differential: ${filterItemObject.description}`;
    case 'language':
      return filterItemObject.custom_description;
    default:
      return filterItemObject.short_description ||
      filterItemObject.description ||
      filterItemObject.long_description ||
      filterItemObject.code ||
      filterItemObject.name ||
      '';
  }
}

// when getting pill descriptions for posts or missions, perform alternate method
export function getPostOrMissionDescription(data) {
  if (data.type === 'post') {
    return `${getPostName(data)}`;
  }
  return false;
}

export function doesCodeOrIdMatch(filterItem, filterItemObject, mappedObject) {
  const filterCode = filterItemObject.code;
  const filterRef = filterItem.item.selectionRef;
  const filterId = filterItemObject.id;

  const codeAndRefMatch = filterCode &&
    filterCode.toString() === mappedObject.codeRef.toString() &&
    filterRef === mappedObject.selectionRef;

  const idAndRefMatch = filterId &&
  filterItemObject.id.toString() === mappedObject.codeRef.toString() &&
  filterRef === mappedObject.selectionRef;

  if (codeAndRefMatch || idAndRefMatch) {
    return true;
  }
  return false;
}

export function isBooleanFilter(description) {
  if (
    description === 'COLA' ||
    description === 'available'
  ) {
    return true;
  }
  return false;
}

export function isPercentageFilter(description) {
  if (
    description === 'dangerPay' ||
    description === 'postDiff'
  ) {
    return true;
  }
  return false;
}
