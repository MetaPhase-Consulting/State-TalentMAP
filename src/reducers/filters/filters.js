import { COMMON_PROPERTIES, ENDPOINT_PARAMS } from '../../Constants/EndpointParams';

// Set what filters we want to fetch
const items =
  {
    filters: [
      {
        item: {
          title: 'Projected Vacancy',
          sort: 50,
          isToggle: true,
          description: 'projectedVacancy',
          selectionRef: ENDPOINT_PARAMS.projectedVacancy,
          text: 'Toggle between available and projected vacancy positions',
          choices: [
          ],
        },
        data: [
          { code: 'open', short_description: 'Open Positions' },
          { code: 'projected', short_description: 'Projected Vacancies' },
        ],
      },
      {
        item: {
          title: 'Bid Cycle',
          sort: 100,
          description: 'bidCycle',
          endpoint: 'bidcycle/?active=true&ordering=name',
          selectionRef: ENDPOINT_PARAMS.bidCycle,
          onlyAvailablePositions: true,
          text: 'Choose Bid Cycles',
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Bid Season',
          sort: 150,
          description: 'bidSeason',
          endpoint: 'fsbid/bid_seasons/',
          selectionRef: ENDPOINT_PARAMS.bidSeason,
          text: 'Choose Bid Seasons',
          onlyProjectedVacancy: true,
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Skill',
          sort: 200,
          description: 'skill',
          endpoint: 'skill/',
          selectionRef: ENDPOINT_PARAMS.skill,
          text: 'Choose Skills',
        },
        data: [
        ],
      },
      {
        // we won't query position directly on cone, but instead
        // use the individual Skill codes
        item: {
          title: 'Skill Cone',
          description: 'skillCone',
          endpoint: 'skill/cone/',
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Language',
          sort: 500,
          description: 'language',
          endpoint: 'language/?is_available=true',
          selectionRef: ENDPOINT_PARAMS.language,
          text: 'Choose languages',
          onlyAvailablePositions: true,
        },
        data: [
        ],
        // Allow users to include languages with no code. This option is not supplied from
        // the endpoint, so we define it here.
        initialData: [
          {
            code: COMMON_PROPERTIES.NULL_LANGUAGE,
            short_description: 'No language requirement',
            custom_description: 'No language requirement',
          },
        ],
      },
      {
        item: {
          title: 'Language',
          sort: 500,
          description: 'language',
          endpoint: 'language/',
          selectionRef: ENDPOINT_PARAMS.language,
          text: 'Choose languages',
          onlyProjectedVacancy: true,
        },
        data: [
        ],
        // Allow users to include languages with no code. This option is not supplied from
        // the endpoint, so we define it here.
        initialData: [
          {
            code: COMMON_PROPERTIES.NULL_LANGUAGE,
            short_description: 'No language requirement',
            custom_description: 'No language requirement',
          },
        ],
      },
      {
        // This is primarily to allow the user to "select all" languages,
        // but could also be extended to group the languages, like we do
        // with skills/cones.
        item: {
          title: 'Language Group',
          altTitle: 'Language',
          description: 'languageGroup',
        },
        data: [
          // useAll means that we don't need to explicitly list the languages here.
          { id: 1, name: 'Languages', isSelected: false, useAll: true },
        ],
      },
      {
        item: {
          title: 'Grade',
          sort: 300,
          description: 'grade',
          endpoint: 'grade/?is_available=true',
          selectionRef: ENDPOINT_PARAMS.grade,
          text: 'Choose grades',
          onlyAvailablePositions: true,
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Grade',
          sort: 300,
          description: 'grade',
          endpoint: 'grade/',
          selectionRef: ENDPOINT_PARAMS.grade,
          text: 'Choose grades',
          onlyProjectedVacancy: true,
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Tour of Duty',
          sort: 400,
          description: 'tod',
          endpoint: 'tour_of_duty/?is_available=true&ordering=months',
          selectionRef: ENDPOINT_PARAMS.tod,
          text: 'Choose Tour of Duty length',
          onlyAvailablePositions: true,
          choices: [
          ],
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Tour of Duty',
          sort: 400,
          description: 'tod',
          endpoint: 'tour_of_duty/?ordering=months',
          selectionRef: ENDPOINT_PARAMS.tod,
          text: 'Choose Tour of Duty length',
          onlyProjectedVacancy: true,
          choices: [
          ],
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Bureau',
          sort: 100,
          description: 'region',
          endpoint: 'organization/?is_bureau=true&is_regional=true',
          selectionRef: ENDPOINT_PARAMS.org,
          text: 'Choose bureau',
          choices: [
          ],
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Functional Bureaus',
          sort: 105,
          description: 'functionalRegion',
          endpoint: 'organization/group/',
          selectionRef: ENDPOINT_PARAMS.functionalOrg,
          text: 'Choose functional bureau',
          choices: [
          ],
        },
        data: [],
      },
      {
        item: {
          title: 'COLA',
          sort: 800,
          bool: true, // use bool: true to share a common HTML template
          description: 'COLA',
          selectionRef: ENDPOINT_PARAMS.cola,
          text: 'Include only positions with COLA',
          choices: [
          ],
        },
        data: [
          { code: '0', short_description: 'Yes' }, // use a code of 0 to specify we want to return results where COLA > 0
        ],
      },
      {
        item: {
          title: 'Post Differential',
          sort: 600,
          description: 'postDiff',
          selectionRef: ENDPOINT_PARAMS.postDiff,
          text: 'Include only positions with a post differential',
          choices: [
          ],
        },
        data: [
          { id: 0, code: '0', description: 'No post differential' },
          { id: 5, code: '5', description: '5%' },
          { id: 10, code: '10', description: '10%' },
          { id: 15, code: '15', description: '15%' },
          { id: 20, code: '20', description: '20%' },
          { id: 25, code: '25', description: '25%' },
          { id: 30, code: '30', description: '30%' },
          { id: 35, code: '35', description: '35%' },
        ],
      },
      {
        item: {
          title: 'Danger Pay',
          sort: 700,
          description: 'dangerPay',
          selectionRef: ENDPOINT_PARAMS.danger,
          text: 'Include only positions with danger pay',
          choices: [
          ],
        },
        data: [
          { id: 100, code: '0', description: 'No danger pay' },
          { id: 150, code: '15', description: '15%' },
          { id: 200, code: '20', description: '20%' },
          { id: 250, code: '25', description: '25%' },
          { id: 350, code: '35', description: '35%' },
        ],
      },
      {
        item: {
          title: 'Domestic',
          sort: 900,
          description: 'domestic',
          selectionRef: ENDPOINT_PARAMS.domestic,
          text: 'Include only domestic positions',
          choices: [
          ],
        },
        data: [
          { code: 'true', short_description: 'Domestic' },
          { code: 'false', short_description: 'Overseas' },
        ],
      },
      {
        item: {
          title: 'Available (No handshakes)',
          sort: 950,
          bool: true,
          description: 'available',
          selectionRef: ENDPOINT_PARAMS.available,
          text: 'Include only available positions',
          choices: [
          ],
        },
        data: [
          { code: 'true', short_description: 'Yes' },
        ],
      },

      /* Currently we don't display this as a filter, but will appear
      as a pill if the query param exists (e.g., the user clicked on Featured positions
      positions from the home page). */
      {
        item: {
          title: 'Featured',
          sort: 1000,
          bool: true,
          description: 'service_needs',
          selectionRef: ENDPOINT_PARAMS.highlighted,
          text: 'Include only featured positions',
          choices: [
          ],
        },
        data: [
          { code: 'true', short_description: 'Featured' },
        ],
      },

      {
        item: {
          title: 'Post',
          altTitle: 'Location',
          sort: 1100,
          bool: false,
          description: 'post',
          endpoint: 'orgpost/?limit=500&is_available=true',
          selectionRef: ENDPOINT_PARAMS.post,
          onlyAvailablePositions: true,
          choices: [
          ],
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Post',
          altTitle: 'Location',
          sort: 1100,
          bool: false,
          description: 'post',
          endpoint: 'orgpost/?limit=500',
          selectionRef: ENDPOINT_PARAMS.post,
          onlyProjectedVacancy: true,
          choices: [
          ],
        },
        data: [
        ],
      },
    ],
  };

export function filtersHasErrored(state = false, action) {
  switch (action.type) {
    case 'FILTERS_HAS_ERRORED':
      return action.hasErrored;
    default:
      return state;
  }
}
export function filtersIsLoading(state = true, action) {
  switch (action.type) {
    case 'FILTERS_IS_LOADING':
      return action.isLoading;
    default:
      return state;
  }
}
export function filters(state = items, action) {
  switch (action.type) {
    case 'FILTERS_FETCH_DATA_SUCCESS':
      return action.filters;
    default:
      return state;
  }
}
