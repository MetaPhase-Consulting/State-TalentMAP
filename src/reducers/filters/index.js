import { filters, filtersAP, filtersHasErrored, filtersIsLoading } from './filters';

export default { filters, filtersAP, filtersHasErrored, filtersIsLoading };
