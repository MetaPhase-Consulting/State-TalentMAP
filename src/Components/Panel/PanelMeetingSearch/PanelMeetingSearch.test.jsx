import TestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJSON from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import PanelMeetingSearch from './PanelMeetingSearch';
import filters from '../../../__mocks__/filtersArray';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('PanelMeetingSearchComponent', () => {
  it('is defined', () => {
    const wrapper = TestUtils.renderIntoDocument(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <PanelMeetingSearch isCDO />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot when CDO', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <PanelMeetingSearch isCDO />
        </MemoryRouter>
      </Provider>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when not CDO', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <PanelMeetingSearch isCDO={false} />
        </MemoryRouter>
      </Provider>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('handles loading', () => {
    const wrapper = TestUtils.renderIntoDocument(
      <Provider store={mockStore({
        filterData: [],
        filtersIsLoading: true,
      })}
      >
        <MemoryRouter>
          <PanelMeetingSearch />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toBeDefined();
  });

  it('handles not loading', () => {
    const wrapper = TestUtils.renderIntoDocument(
      <Provider store={mockStore({
        filterData: filters,
        filtersIsLoading: false,
      })}
      >
        <MemoryRouter>
          <PanelMeetingSearch />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toBeDefined();
  });
});
