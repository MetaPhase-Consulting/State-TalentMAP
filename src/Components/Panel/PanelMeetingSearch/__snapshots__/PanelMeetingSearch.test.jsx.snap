// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeAgendaSearchComponent matches snapshot when CDO 1`] = `
<div>
  <div
    className="panel-meeting-search-page"
  >
    <div
      className="panel-meeting-search-upper-section"
    >
      <div
        className="results-search-bar results-single-search"
      >
        <div
          className="usa-grid-full results-search-bar-container"
        >
          <ProfileSectionTitle
            icon="comment"
            title="Panel Meeting Search"
          />
          <ForwardRef
            defaultValue=""
            label="Search for a Panel Meeting"
            onChange={[Function]}
            placeHolder="Search using Panel ID"
            submitSearch={[Function]}
            textSearch=""
          />
          <div
            className="filterby-container"
          >
            <div
              className="filterby-label"
            >
              Filter by:
            </div>
            <div
              className="filterby-clear"
            />
          </div>
          <div
            className="usa-width-one-whole panel-meeting-search-filters"
          >
            <div
              className="filter-div"
            >
              <div
                className="label"
              >
                Type:
              </div>
              <r
                dropdownHeight={255}
                filterDebounce={150}
                id="picky"
                includeFilter={true}
                includeSelectAll={true}
                keepOpen={true}
                labelKey="description"
                multiple={true}
                numberDisplayed={2}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "code": "ID",
                      "description": "ID",
                    },
                    Object {
                      "code": "ML",
                      "description": "ML",
                    },
                  ]
                }
                placeholder="Select Meeting Type"
                renderList={[Function]}
                selectAllMode="default"
                selectAllText="Select all"
                tabIndex={0}
                value={Array []}
                valueKey="code"
              />
            </div>
            <div
              className="filter-div"
            >
              <div
                className="label label-date"
              >
                Meeting Date:
              </div>
              <DateRangePicker
                calendarIcon={null}
                clearIcon={
                  <svg
                    className="react-daterange-picker__clear-button__icon react-daterange-picker__button__icon"
                    height={19}
                    stroke="black"
                    strokeWidth={2}
                    viewBox="0 0 19 19"
                    width={19}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <line
                      x1="4"
                      x2="15"
                      y1="4"
                      y2="15"
                    />
                    <line
                      x1="15"
                      x2="4"
                      y1="4"
                      y2="15"
                    />
                  </svg>
                }
                closeCalendar={true}
                isOpen={null}
                maxDetail="month"
                name="daterange"
                onChange={[Function]}
                openCalendarOnFocus={true}
                rangeDivider="–"
                showLeadingZeros={true}
                value={null}
              />
            </div>
            <div
              className="filter-div"
            >
              <div
                className="label"
              >
                Status:
              </div>
              <r
                dropdownHeight={255}
                filterDebounce={150}
                id="picky"
                includeFilter={true}
                includeSelectAll={true}
                keepOpen={true}
                labelKey="description"
                multiple={true}
                numberDisplayed={2}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "code": "initiated",
                      "description": "Initiated",
                    },
                    Object {
                      "code": "addendum",
                      "description": "Addendum",
                    },
                    Object {
                      "code": "post_panel",
                      "description": "Post Panel",
                    },
                  ]
                }
                placeholder="Select Meeting Status"
                renderList={[Function]}
                selectAllMode="default"
                selectAllText="Select all"
                tabIndex={0}
                value={Array []}
                valueKey="code"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmployeeAgendaSearchComponent matches snapshot when not CDO 1`] = `
<div>
  <div
    className="panel-meeting-search-page"
  >
    <div
      className="panel-meeting-search-upper-section"
    >
      <div
        className="results-search-bar results-single-search"
      >
        <div
          className="usa-grid-full results-search-bar-container"
        >
          <ProfileSectionTitle
            icon="comment"
            title="Panel Meeting Search"
          />
          <ForwardRef
            defaultValue=""
            label="Search for a Panel Meeting"
            onChange={[Function]}
            placeHolder="Search using Panel ID"
            submitSearch={[Function]}
            textSearch=""
          />
          <div
            className="filterby-container"
          >
            <div
              className="filterby-label"
            >
              Filter by:
            </div>
            <div
              className="filterby-clear"
            />
          </div>
          <div
            className="usa-width-one-whole panel-meeting-search-filters"
          >
            <div
              className="filter-div"
            >
              <div
                className="label"
              >
                Type:
              </div>
              <r
                dropdownHeight={255}
                filterDebounce={150}
                id="picky"
                includeFilter={true}
                includeSelectAll={true}
                keepOpen={true}
                labelKey="description"
                multiple={true}
                numberDisplayed={2}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "code": "ID",
                      "description": "ID",
                    },
                    Object {
                      "code": "ML",
                      "description": "ML",
                    },
                  ]
                }
                placeholder="Select Meeting Type"
                renderList={[Function]}
                selectAllMode="default"
                selectAllText="Select all"
                tabIndex={0}
                value={Array []}
                valueKey="code"
              />
            </div>
            <div
              className="filter-div"
            >
              <div
                className="label label-date"
              >
                Meeting Date:
              </div>
              <DateRangePicker
                calendarIcon={null}
                clearIcon={
                  <svg
                    className="react-daterange-picker__clear-button__icon react-daterange-picker__button__icon"
                    height={19}
                    stroke="black"
                    strokeWidth={2}
                    viewBox="0 0 19 19"
                    width={19}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <line
                      x1="4"
                      x2="15"
                      y1="4"
                      y2="15"
                    />
                    <line
                      x1="15"
                      x2="4"
                      y1="4"
                      y2="15"
                    />
                  </svg>
                }
                closeCalendar={true}
                isOpen={null}
                maxDetail="month"
                name="daterange"
                onChange={[Function]}
                openCalendarOnFocus={true}
                rangeDivider="–"
                showLeadingZeros={true}
                value={null}
              />
            </div>
            <div
              className="filter-div"
            >
              <div
                className="label"
              >
                Status:
              </div>
              <r
                dropdownHeight={255}
                filterDebounce={150}
                id="picky"
                includeFilter={true}
                includeSelectAll={true}
                keepOpen={true}
                labelKey="description"
                multiple={true}
                numberDisplayed={2}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "code": "initiated",
                      "description": "Initiated",
                    },
                    Object {
                      "code": "addendum",
                      "description": "Addendum",
                    },
                    Object {
                      "code": "post_panel",
                      "description": "Post Panel",
                    },
                  ]
                }
                placeholder="Select Meeting Status"
                renderList={[Function]}
                selectAllMode="default"
                selectAllText="Select all"
                tabIndex={0}
                value={Array []}
                valueKey="code"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
